### DISCUSSION TITLE
Why can't I display the Archive and Categories layouts by default when I use the Built-in blog plugin?

### DISCUSSION THREAD

**Discussion by jygzyc** [squidfunk/mkdocs-material#6542] [Q&A] [answered]
When I use the Built-in blog plugin with the following configuration, the sidebar layout is empty. Could there be an issue with my settings?
```yml
#plugins
plugins:
  - blog:
        blog_dir: blog
        post_url_date_format: yyyy/MM
        post_url_format: "{date}/{file}"
        archive: true
        categories: true
```

![image](https://github.com/squidfunk/mkdocs-material/assets/33087260/ccba2d85-b772-440b-bb51-73c889691f62)

What can I do to restore the expected behavior?

![image](https://github.com/squidfunk/mkdocs-material/assets/33087260/b76a92a3-a9f2-43c4-a325-e41fd646371b)

**Comment by alexvoss** [comment]
@squidfunk, this is odd. I can replicate the problem by using the [blog-basic example](https://github.com/mkdocs-material/examples/tree/master/examples/blog-basic), which [built just fine when it was deployed to GitHub Pages](https://mkdocs-material.github.io/examples/blog-basic/). I tried various versions (latest Insiders from GitHub as well as the latest public release). I still hope this is a configuration error but it seems that at least two people on this planet managed to produce it.

Somewhat unrelated: I tried to build the blog-basic example as part of the whole set of examples - and it crashed my machine. It seems to run into endless forking? (I seem to remember there were some recent changes to the logic and some people reporting issues in Docker?) So, at the moment I am only able to say that it does not produce the archive and categories navigation when used stand-alone.

**Reply by jygzyc** [reply]
@alexvoss When I deploy to GitHub Pages, I still encounter the issue as shown in the figure above. The link to my site is as follows: 
[Ecool Notes](https://note.lilac.fun/blog/)

**Reply by squidfunk** [reply]
Please open a new issue, discussions are not meant for reporting bugs.

> Somewhat unrelated: I tried to build the blog-basic example as part of the whole set of examples - and it crashed my machine. It seems to run into endless forking? 

Please upgrade Insiders. This is likely related to the latest errors fixed in the projects plugin.

**Reply by alexvoss** [reply]
Sorry, I did not mean to imply that the GitHub deployment is the issue here, just to say that I am observing differences in behaviour between what happens on my machine today and what obviously worked in the past. I am trying to narrow down where this is coming from but  ran into other issues while doing this. It might help if you could produce a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) as well so we can investigate this.

**Reply by squidfunk** [reply]
Other than that I'm not sure why you see this error. Running the blog basic example on the latest Insiders I see:

<img width="1018" alt="Bildschirm­foto 2023-12-19 um 16 22 51" src="https://github.com/squidfunk/mkdocs-material/assets/932156/1ebaba01-722f-4df1-a3f8-7a1fa7eb9617">

I'm gonna need a minimal reproduction that showcases the error. It is very likely a configuration error.

**Reply by alexvoss** [reply]
Ok, get the missing index pages on both Linux and MacOS (installed Sonoma the other day, in case that turns out to matter). I scrapped the venv on Linux and re-installed using:

```
pip install git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git
``` 

Same result:

![image](https://github.com/squidfunk/mkdocs-material/assets/4134224/194e54d2-cef1-4c92-b5f8-96db05d68aac)

There are some other differences in the rendering, perhaps they give a hint as to what is going on? Did you build the example stand-alone or as part of the larger build with the projects plugin?

**Reply by squidfunk** [reply]
Both. Standalone, as well as with the projects plugin, same result. The only difference is that `navigation.sections` is only enabled for the projects plugin, not in standalone mode.

**Reply by squidfunk** [reply]
... and I just confirmed that building from inside Docker works as well, both for the project mode and non-project mode, so Linux doesn't seem to be a problem as well, at least on my machine.

**Reply by jygzyc** [reply]
@squidfunk I just tested it and found that when I add the following configuration to the [blog-basic example](https://github.com/mkdocs-material/examples/tree/master/examples/blog-basic),  the navigation disappears:
![mkdocs.yml](https://github.com/squidfunk/mkdocs-material/assets/33087260/af3ebd9a-0b35-4a36-9431-3699179ddf6f)
![Navigation disappearing](https://github.com/squidfunk/mkdocs-material/assets/33087260/f54f1907-1884-4f78-8940-fe7120d72ff7)

**Reply by jygzyc** [reply]
By the way, this is my current version：
![image](https://github.com/squidfunk/mkdocs-material/assets/33087260/61052140-83bf-4eec-b4df-be07f375d0d1)

**Reply by squidfunk** [reply]
Yes, `blog: index.md` is incorrect. The blog __must__ be a section:

``` yaml
nav:
  - Blog:
    -  index.md
```

**Reply by jygzyc** [reply]
@squidfunk  However, when I build with the "documentation+blog" mode, shouldn't the entry point of the blog be at "blog/index.md" instead?
![image](https://github.com/squidfunk/mkdocs-material/assets/33087260/cb41e1ee-76c0-4e3f-9a99-dd250226d315)

**Reply by alexvoss** [reply]
> The blog must be a section

Aha, that fixes it for me. I looked at the blog documentation today and was specifically looking for the instructions for setting up `nav:` but could not find it. Did I miss it?

**Reply by squidfunk** [reply]
[It's currently only mentioned here](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-pages) at the moment. It's also why I mentioned countless times that we desperately need to rework the blog setup guide and provide starters and examples because many users seem to have problems setting things up 😉 As you're currently building a blog starter template, reworking the blog setup guide is something that would provide immense value, because it will help us bring down the number of opened discussions and allow us to provide a better DX.

**Reply by squidfunk** [reply]
I'm currently focusing my work on tags, projects and search. I will not be able to rework the documentation in the next coming months, so I need to rely on the help of contributors. The tags plugin will also need reworked documentation once it's stable, because the possibilities will be insane compared to now.

**Reply by alexvoss** [reply]
Ok, I will do a PR for the documentation and also fix up the example to show how navigation should be configured.

**Reply by alexvoss** [reply, marked as answer]
The [update to the documentation is now published.](https://squidfunk.github.io/mkdocs-material/plugins/blog/#navigation). Hope this makes it easier to get started with blogs.
